src/debug_clang.h

```diff
@@ -0,0 +1,85 @@
+#ifndef JUCI_DEBUG_CLANG_H_
+#define JUCI_DEBUG_CLANG_H_
+
+#include <boost/filesystem.hpp>
+#include <unordered_map>
+#include <lldb/API/SBDebugger.h>
+#include <lldb/API/SBListener.h>
+#include <lldb/API/SBProcess.h>
+#include <thread>
+#include <mutex>
+
+namespace Debug {
+  class Clang {
```

Comment:
The source implementation is still in source*, so this is where syntax highlighting and completion for instance go (subclass Source::View like before). The project class is needed since you might have several file types related to a language, and one needs to figure out the project root path (even without a source file opened), so in order to implement a new project for a programming language with run and debug functions you subclass Project::Language. See https://github.com/eidheim/jucipp/blob/master/src/project.cc#L528-L542 for instance for python and javascript implementations for run. These are quite simple yet though. Javascript project implementation for instance need to search for the package.json file.